examples/ChatQnA/deploy/xeon.md

```diff
@@ -0,0 +1,875 @@
+
+```
+
+### TEI Reranking Service
+
+The TEI Reranking Service reranks the documents returned by the retrieval
+service. It consumes the query and list of documents and returns the document
+index based on decreasing order of the similarity score. The document
+corresponding to the returned index with the highest score is the most relevant
+document for the input query.
+```
+curl http://${host_ip}:8808/rerank \
+    -X POST \
+    -d '{"query":"What is Deep Learning?", "texts": ["Deep Learning is not...", "Deep learning is..."]}' \
+    -H 'Content-Type: application/json'
+```
+
+Output is:  `[{"index":1,"score":0.9988041},{"index":0,"score":0.022948774}]`
+
+
+### Reranking Microservice
+
+
+The reranking microservice consumes the TEI Reranking service and pads the
+response with default parameters required for the llm microservice.
+
+```
+curl http://${host_ip}:8000/v1/reranking\
+  -X POST \
+  -d '{"initial_query":"What is Deep Learning?", "retrieved_docs": \
+     [{"text":"Deep Learning is not..."}, {"text":"Deep learning is..."}]}' \
+  -H 'Content-Type: application/json'
+```
+
+The input to the microservice is the `initial_query` and a list of retrieved
+documents and it outputs the most relevant document to the initial query along
+with other default parameter such as temperature, `repetition_penalty`,
+`chat_template` and so on. We can also get top n documents by setting `top_n` as one
+of the input parameters. For example:
+
+```
+curl http://${host_ip}:8000/v1/reranking\
+  -X POST \
+  -d '{"initial_query":"What is Deep Learning?" ,"top_n":2, "retrieved_docs": \
+     [{"text":"Deep Learning is not..."}, {"text":"Deep learning is..."}]}' \
+  -H 'Content-Type: application/json'
+```
+
+Here is the output:
+
+```
+{"id":"e1eb0e44f56059fc01aa0334b1dac313","query":"Human: Answer the question based only on the following context:\n    Deep learning is...\n    Question: What is Deep Learning?","max_new_tokens":1024,"top_k":10,"top_p":0.95,"typical_p":0.95,"temperature":0.01,"repetition_penalty":1.03,"streaming":true}
+
+```
+You may notice reranking microservice are with state ('ID' and other meta data),
+while reranking service are not.
+
+### vLLM and TGI Service
+
+::::{tab-set}
+
+:::{tab-item} vllm
+:sync: vllm
+
+```
+curl http://${host_ip}:9009/v1/completions \
+  -H "Content-Type: application/json" \
+  -d '{"model": "Intel/neural-chat-7b-v3-3", \
+     "prompt": "What is Deep Learning?", \
+     "max_tokens": 32, "temperature": 0}'
+```
+
+vLLM service generate text for the input prompt. Here is the expected result
+from vllm:
+
+```
+{"generated_text":"We have all heard the buzzword, but our understanding of it is still growing. It’s a sub-field of Machine Learning, and it’s the cornerstone of today’s Machine Learning breakthroughs.\n\nDeep Learning makes machines act more like humans through their ability to generalize from very large"}
+```
+
+**NOTE**: After launch the vLLM, it takes few minutes for vLLM server to load

```

Comment:
After launching the vLLM service it takes a few minutes for the vLLM server to load